src/stk/molecular/topology_graphs/cof/periodic_hexagonal.py

```diff
@@ -0,0 +1,165 @@
+
+        lattice_size : :class:`tuple` of :class:`int`
+            The size of the lattice in the x, y and z directions.
+
+        vertex_alignments : :class:`dict`, optional
+            A mapping from the id of a :class:`.Vertex`
+            to an :class:`.Edge` connected to it.
+            The :class:`.Edge` is used to align the first
+            :class:`.FunctionalGroup` of a :class:`.BuildingBlock`
+            placed on that vertex. Only vertices which need to have
+            their default edge changed need to be present in the
+            :class:`dict`. If ``None`` then the default edge is used
+            for each vertex. Changing which :class:`.Edge` is used will
+            mean that the topology graph represents different
+            structural isomers. The edge is referred to by a number
+            between ``0`` (inclusive) and the number of edges the
+            vertex is connected to (exclusive).
+
+        reaction_factory : :class:`.ReactionFactory`, optional
+            The reaction factory to use for creating bonds between
+            building blocks.
+
+        num_processes : :class:`int`, optional
+            The number of parallel processes to create during
+            :meth:`construct`.
+
+        Raises
+        ------
+        :class:`AssertionError`
+            If the any building block does not have a
+            valid number of functional groups.
+
+        :class:`ValueError`
+            If the there are multiple building blocks with the
+            same number of functional_groups in `building_blocks`,
+            and they are not explicitly assigned to vertices. The
+            desired placement of building blocks is ambiguous in
+            this case.
+
+        :class:`~.cof.UnoccupiedVertexError`
+            If a vertex of the COF topology graph does not have a
+            building block placed on it.
+
+        :class:`~.cof.OverlyOccupiedVertexError`
+            If a vertex of the COF topology graph has more than one
+            building block placed on it.
+
+        """
+
+        self._internal = Hexagonal(
+            building_blocks=building_blocks,
+            lattice_size=lattice_size,
+            periodic=True,
+            vertex_alignments=vertex_alignments,
+            reaction_factory=reaction_factory,
+            num_processes=num_processes,
+        )
+
+    def construct(self):
+        """
+        Construct a :class:`.ConstructedMolecule`.
+
+        Returns
+        -------
+        :class:`.ConstructionResult`
+            The data describing the :class:`.ConstructedMolecule`.
+
+        """
+
+        return self._internal.construct()
+
+    def get_building_blocks(self):
+        yield from self._internal.get_building_blocks()
+
+    def get_num_building_block(self, building_block):
+        return self._internal.get_num_building_block(building_block)
+
+    def clone(self):
+        return self._internal.clone()
+
+    def get_periodic_cell(self):
```

Comment:
What do you think about this being `get_periodic_info` and returning a `PeriodicInfo` instance?

Reply:
Done after discussion.